src/index.js

```diff
@@ -3,15 +3,18 @@ import ReactDOM from "react-dom";
 import "./App.css";
 import App from "./App";
 import { makeServer } from "./server";
-import { BrowserRouter as Router} from 'react-router-dom';
+import { BrowserRouter as Router } from "react-router-dom";
+import { FilterProvider } from "./context/FilterContext";
 
 // Call make Server
 makeServer();
 
 ReactDOM.render(
   <React.StrictMode>
     <Router>
-      <App />
+      <FilterProvider>
+        <App />
+      </FilterProvider>
```

Comment:
I am not sure why are you using <App /> two times here.
I think all the context providers should be used in App.js page. The index.js page should look clean.